packages/core-mobile/app/new/features/track/screens/TrackTokenDetailScreen.tsx

```diff
@@ -178,13 +193,19 @@ const TrackTokenDetailScreen = (): JSX.Element => {
     }
   }, [openUrl, tokenInfo?.urlHostname, back])
 
-  const handleBuy = useCallback((): void => {
-    navigate({
-      // @ts-ignore TODO: make routes typesafe
-      pathname: '/buy',
-      params: { showAvaxWarning: 'true' }
-    })
-  }, [navigate])
+  const handleBuy = useCallback(
+    (contractAddress?: string): void => {
+      const cryptoCurrency = getBuyableCryptoCurrency({
+        cryptoCurrencies,
+        tokenOrAddress: contractAddress
+      })
+      navigateToBuy({
+        showAvaxWarning: true,
+        cryptoCurrency
+      })
```

Comment:
same here. I think it will be better to have `navigateToBuy( pass in token or address here )`. that way the meld logic is abstracted away.